### PR TITLE
Fix SWIG preprocessor issue with VOCABs in enums

### PR DESCRIPTION
**Pull request by jgvictores**
- SWIG was generating compile errors on `VOCAB`s in `enums`s starting by 'b'/'c'/'d'.
- d7b4099 introduced a hack to avoid SWIG binding compilation errors by substituting `char`s by equivalent `int`s.
- A deep issue was buried :skull:: only `VOCAB`s with that hack were correctly formed.
- In reality, all other `VOCAB`s generated a **copy** of the first `char`, rather than the correct `int` (e.g. `VOCAB_PIXEL_ENCODING_BAYER_GBRG16=VOCAB4('g','b','1','6');` generated `'g'` rather than `909206119`).
- Note this only affected the `VOCAB` in `enum` with SWIG combination.
- TLDR: roboticslab-uc3m/kinematics-dynamics#150 [ where @PeterBowman proposed the `constexpr` solution]
- PR also removes obsolete hacks.

Updates:
- Related: https://github.com/swig/swig/issues/737
- Introduced `SWIG_PREPROCESSOR_SHOULD_SKIP_THIS` to preserve original c-style macro VOCAB.

### CODE REVIEW

**Comment by jgvictores** [comment]
Since Appveyor was having issues such as [these](https://ci.appveyor.com/project/robotology/yarp/build/appveyor.4293/job/2ar65e6wads9xsq4), I thought it was reasonable to revert the `constexpr` to the c-style macro, and fix it on the SWIG side as it is SWIG-specific (`SWIG_PREPROCESSOR_SHOULD_SKIP_THIS` inspired on the doxygen analogous) at 6565ffbe4f9b647ea796f51dcb93625c6a651335.

Everything working (`int`s correctly generated, checked via Python), reasonably stylish. I cannot change the branch name, but that shouldn't be kept in the historic anyway.

**Comment by jgvictores** [comment]
@traversaro Any idea of which release notes should be touched?

**Comment by traversaro** [comment]
This document: https://github.com/robotology/yarp/blob/master/doc/release/v2_3_72_1.md .

**Comment by jgvictores** [comment]
Okay, fixed. A bit of `--force push`, correct release notes updated at 4b49f64. Good to go!

**Comment by traversaro** [comment]
Given that we playing a bit with rebase, I think that https://github.com/robotology/yarp/pull/1655/commits/94f2f08dfddce2ada3b69ebc1cecdcf649bb3641 and https://github.com/robotology/yarp/pull/1655/commits/6565ffbe4f9b647ea796f51dcb93625c6a651335 can be merged together, and moved in history before https://github.com/robotology/yarp/pull/1655/commits/b51ac47af512d0f494aa4264242f1ee355e75d54 .

**Comment by jgvictores** [comment]
> To avoid regression in the future, it would be nice to have at least a regression test for that.

Done at 0f66a46, updated release notes at bc068f7.

Note commit message on 0f66a46:
- The first test is not sufficient, as `yarp::Vocab::encode()/decode()` were working, as well as their corresponding bindings.
- Thus, a second test is introduced. It catches and tests some representative YARP `VOCAB`s defined in `enums`, essentially some from the few that are exposed, that can be "seen" by Lua, and thus evaluated. This is actually the test that fails with the old YARP code, and is fixed by this PR.

**Comment by drdanz** [comment]
Merged, thanks!
